tests/ut_MeterExec.cpp

```diff
@@ -0,0 +1,201 @@
+#include "gtest/gtest.h"
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+
+#include "Options.hpp"
+#include "protocols/MeterExec.hpp"
+
+// this is a dirty hack. we should think about better ways/rules to link against the
+// test objects.
+// e.g. single files in the tests directory that just include the real file. This way one
+// will avoid multiple includes of the same file in a different test case
+// (or linking against the real file from the CMakeLists.txt)
+
+//#include "../src/protocols/MeterExec.cpp"
+
+int writes(int fd, const char *str);
+
+TEST(MeterExec, basic) {
+	char tempfilename[L_tmpnam+1];
+	ASSERT_NE(tmpnam_r(tempfilename), (char*)0);
```

Comment:
you will need to create a file that can be executed. E.g. some shell script that returns the values (#!/bin/bash echo x y...)